src/FiberMap.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace React\Async;
+
+use Fiber;
+use React\Promise\PromiseInterface;
+
+/**
+ * @internal
+ */
+final class FiberMap
+{
+    private array $status = [];
+    private array $map   = [];
+
+    public function register(Fiber $fiber): void
+    {
+        $this->status[spl_object_hash($fiber)] = false;
+        $this->map[spl_object_hash($fiber)] = [];
+    }
+
+    public function cancel(Fiber $fiber): void
+    {
+        $this->status[spl_object_hash($fiber)] = true;
+    }
+
+    public function isCanceled(Fiber $fiber): bool
+    {
+        return $this->status[spl_object_hash($fiber)];
+    }
+
+    public function attachPromise(Fiber $fiber, PromiseInterface $promise): void
+    {
+        $this->map[spl_object_hash($fiber)][spl_object_hash($promise)] = $promise;
+    }
+
+    public function has(Fiber $fiber): bool
+    {
+        return array_key_exists(spl_object_hash($fiber), $this->map);
+    }
+
+    public function getPromises(Fiber $fiber): array
+    {
+        return $this->map[spl_object_hash($fiber)];
```

Comment:
Not sure this can ever happen, but maybe make sure it doesn't trigger warnings for undefined array keys?

```suggestion
        return $this->map[spl_object_hash($fiber)] ?? [];
```

Reply:
It really shouldn't. But included it just in case

Reply:
It really shouldn't. But included it just in case